samples/uploadDirectoryWithTransferManager.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * @experimental
+ */
+
+// sample-metadata:
+//   title: Upload Folder With Transfer Manager
+//   description: Uploads a folder in parallel utilizing transfer manager.
+//   usage: node uploadFolderWithTransferManager.js <BUCKET_NAME> <DIRECTORY_NAME>
+
+function main(bucketName = 'my-bucket', directoryName = 'my-directory') {
+  // [START storage_upload_directory_transfer_manager]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The local directory to upload
+  // const directoryName = 'your-directory';
+
+  // Imports the Google Cloud client library
+  const {Storage, TransferManager} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  // Creates a transfer manager instance
+  const transferManager = new TransferManager(storage.bucket(bucketName));
```

Comment:
What are your thoughts on accepting a string in TransferManager's constructor and creating a bucket from it?

Reply:
I like this idea. Let me tackle it in a separate PR as to not muddy the waters on this one.